pkg/codegen/nodejs/gen.go

```diff
@@ -1022,12 +1035,21 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) (resourceFil
 				}
 			}
 		}
-		fmt.Fprintf(w, "        }, this);\n")
+		fmt.Fprintf(w, "        }, this")
+
+		if doReturnPlainType {
+			prop := camel("res")
+			fmt.Fprintf(w, ", {plainResourceField: %q}", prop)
```

Comment:
This shouldn't be called "plainResourceField" since the returned value can now be not a resource but any result really.